src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2013,30 +2036,40 @@ protected override void BeginProcessing()
                 }
             }
 
-            if (Wait.IsPresent)
+            if (Wait.IsPresent || _timeoutSpecified)
             {
                 if (process != null)
                 {
                     if (!process.HasExited)
                     {
 #if UNIX
```

Comment:
We should remove two code paths and use `public bool WaitForExit(int milliseconds);` for all plaforms.

Reply:
I'm sorry, I don't quite understand what you are asking here. Are you saying to remove the `ProcessCollection` / `jobObject` and just have a `WaitForExit` for all platforms? e.g.

```
if (!process.WaitForExit(ExitTimeout))
{
    StopProcessOnTimeout(process);
}
```

From what I can see, the Windows specific code keeps track of any child processes that are created, which doesn't happen for UNIX. I am unsure of the reasoning behind this, as the earliest commit on GitHub for these lines does not contain any helpful information.